source/grid/grid_in.cc

```diff
@@ -757,6 +757,31 @@ void GridIn<dim, spacedim>::read_ucd (std::istream &in)
   tria->create_triangulation_compatibility (vertices, cells, subcelldata);
 }
 
+namespace
+{
+  // Forward declaration of helper class
+  template <int dim>
+  class Abaqus_to_UCD;
```

Comment:
This is slightly odd -- you're providing a forward-declaration here and provide the implementation below. Why not just copy the entire implementation here?